nml/actions/action2var.py

```diff
@@ -557,6 +571,38 @@ def parse_via_actionD(self, expr):
     def parse_proc_call(self, expr):
         assert isinstance(expr, expression.SpriteGroupRef)
         var_access = VarAction2ProcCallVar(expr)
+        target = action2.resolve_spritegroup(expr.name)
+        refs = expr.collect_references()
+
+        # Fill param registers for the call
+        tmp_vars = []
+        for i, param in enumerate(expr.param_list):
+            if i > 0: # No operator before first param as per advanced VarAct2 syntax
+                self.var_list.append(nmlop.VAL2)
+                self.var_list_size += 1
+            if refs != [expr]:
+                # For f(x, g(y)), x can be overwritten by y if f and g share the same param registers
+                # Use temporary variables as an intermediate step
+                store_tmp = VarAction2StoreTempVar()
+                tmp_vars.append((store_tmp, VarAction2StoreCallParam(target.register_map[self.feature][i])))
+            else:
+                store_tmp = VarAction2StoreCallParam(target.register_map[self.feature][i])
+            self.parse_expr(reduce_varaction2_expr(param, self.feature))
+            self.var_list.append(nmlop.STO_TMP)
+            self.var_list.append(store_tmp)
```

Comment:
This could be `self.var_list += (nmlop.STO_TMP, store_tmp)`, and similarly below. Current way might be clearer, not sure.